config/src/providers/remappings.rs

```diff
@@ -116,20 +150,24 @@ impl<'a> RemappingsProvider<'a> {
                 insert_closest(&mut lib_remappings, r.context, r.name, r.path.into());
             }
 
-            new_remappings.extend(lib_remappings.into_iter().flat_map(|(context, remappings)| {
-                remappings.into_iter().map(move |(name, path)| Remapping {
-                    context: context.clone(),
-                    name,
-                    path: path.to_string_lossy().into(),
-                })
-            }));
+            new_remappings.extend(
+                lib_remappings
+                    .into_iter()
+                    .flat_map(|(context, remappings)| {
+                        remappings.into_iter().map(move |(name, path)| Remapping {
+                            context: context.clone(),
+                            name,
+                            path: path.to_string_lossy().into(),
+                        })
+                    })
+                    .collect(),
+            );
         }
 
         // remove duplicates at this point
-        new_remappings.sort_by(|a, b| (&a.context, &a.name).cmp(&(&b.context, &b.name)));
-        new_remappings.dedup_by(|a, b| (&a.context, &a.name).eq(&(&b.context, &b.name)));
+        new_remappings.remappings.dedup_by(|a, b| (&a.context, &a.name).eq(&(&b.context, &b.name)));
```

Comment:
why do we need to dedup here?
dedup will only work if sorted, which we're no longer doing.
the new type is supposed to handle this